services/maintenance-page/docker-compose.yml.j2

```diff
@@ -0,0 +1,62 @@
+version: '3.8'
+configs:
+  maintenance_page_api_html:
+    file: ./page/api/index.html
+  nginx_config:
+    file: ./nginx-config/default.conf
+{% for item in DEPLOYMENT_FQDNS.split(",") + [MACHINE_FQDN] %}
+  {{ "maintenance_page_web_" + item.replace('@','').replace(' ','').replace('.','').replace('-','').replace('\'','') + "_html"}}:
+    file: {{ "/".join(REPO_CONFIG_LOCATION.split("/")[:-1]) + "/maintenancePageWebHTML/" + item.replace(" ","") + ".html" }}{% endfor %}
+services:
+{% for j2item in DEPLOYMENT_FQDNS.split(",") + [MACHINE_FQDN] %}
+  maintenance_page_{{j2item.replace('@','').replace(' ','').replace('.','').replace('-','').replace('\'','')}}:
+    configs:
+      - source: {{"maintenance_page_web_" + j2item.replace('@','').replace(' ','').replace('.','').replace('-','').replace('\'','') + "_html"}}
+        target: /usr/share/nginx/html/index.html
+      - source: nginx_config
+        target: /etc/nginx/conf.d/default.conf
+    # nginx config
+    image: nginx:1.23.3
+    networks:
+      - public
+      - monitored
+    deploy:
+      labels:
+        - traefik.enable=true
+        - traefik.docker.network=${PUBLIC_NETWORK}
+        - traefik.http.routers.nginx_web.priority={{MAINTENANCE_PAGES_TRAEFIK_PRIORITY}}
+        - traefik.http.routers.nginx_web.rule={{  "Host(`" + j2item + "`)" }}}
+        - traefik.http.routers.nginx_web.rule=(Host(`{{j2item}}`) && PathPrefix(`/`)) || (HostRegexp(`services.{{j2item}}`,`{subhost:[a-zA-Z0-9-]+}.services.{{j2item}}`) && PathPrefix(`/`)) || (HostRegexp(`services.testing.{{j2item}}`,`{subhost:[a-zA-Z0-9-]+}.services.testing.{{j2item}}`) && PathPrefix(`/`))
+        - traefik.http.routers.nginx_web.tls=true
+        - traefik.http.routers.nginx_web.tls.certresolver=myresolver
+        - traefik.http.services.nginx_web.loadbalancer.server.port=80
+        - traefik.http.routers.nginx_web.entrypoints=https
+{% endfor %}
+  maintenance_page_api:
+    # nginx config
+    image: nginx:1.23.3
+    configs:
+      - source: maintenance_page_api_html
+        target: /usr/share/nginx/html/index.html
```

Comment:
If i understand that well, this responds with an html page for the AP as wellI? 
That is not strictly necessary (might be ok to "replace" the swagger view). 
The API server is used to "talk to client machines". The important thing for the API is to return 503 (service unavailable) with a maintenance message and include in the headers the time to retry (e.g. end of maintenance). 

Please check in the ops issues.  There you should find all explanations.

Reply:
@Surfict created the API status page, he might be able to give more details. I was not introduced to the design ideas behind the API maintenance page, I guess its creation was requested and speced by you @pcrespov .

Please comment here @Surfict

Reply:
The html file returns the following JSON code : 
`{
    "errors": {
        "msg":  "Service currently unavailable due to scheduled maintenance.",
        "type": "runtime_error"
     }
}`
I don't think anything has been done for the headers though.